src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationSource.cs

```diff
@@ -29,22 +29,22 @@ public class DaprSecretStoreConfigurationSource : IConfigurationSource
         /// <summary>
         /// Gets or sets the custom key delimiters. Contains the '__' delimiter by default.
         /// </summary>
-        public IList<string> KeyDelimiters { get; set; } = new List<string> { "__" };
+        public IList<string>? KeyDelimiters { get; set; } = new List<string> { "__" };
 
         /// <summary>
         /// Gets or sets the secret descriptors.
         /// </summary>
-        public IEnumerable<DaprSecretDescriptor> SecretDescriptors { get; set; }
+        public IEnumerable<DaprSecretDescriptor>? SecretDescriptors { get; set; }
 
         /// <summary>
         /// A collection of metadata key-value pairs that will be provided to the secret store. The valid metadata keys and values are determined by the type of secret store used.
         /// </summary>
-        public IReadOnlyDictionary<string, string> Metadata { get; set; }
+        public IReadOnlyDictionary<string, string>? Metadata { get; set; }
 
         /// <summary>
         /// Gets or sets the http client.
         /// </summary>
-        public DaprClient Client { get; set; }
+        public DaprClient? Client { get; set; }
```

Comment:
Most of the properties in this class fall into the category of *should not be null in common usage*. For instance users should not encounter a case where `Client` is null, nor should they set it to null. 

I'd suggest using `{ get; set; } = default!;` For this one.